src/DynamoCoreWpf/ViewModels/Core/GalleryViewModel.cs

```diff
@@ -0,0 +1,177 @@
+            }
+            catch (Exception)
+            {
+                return new GalleryContents();
+            }
+        }
+    }
+
+    public class GalleryViewModel: ViewModelBase
+    {
+        #region public members
+        public string CurrentImagePath { get { return (currentContent == null) ? string.Empty : currentContent.ImagePath; } }
+        public string CurrentHeader { get { return (currentContent == null) ? string.Empty : currentContent.Header; } }
+        public string CurrentBody { get { return (currentContent == null) ? string.Empty : currentContent.Body; } }
+        public string DynamoVersion { get; private set; }
+        public IEnumerable<GalleryContent> Contents { get { return contents; } }
+        public DelegateCommand MoveNextCommand { get; set; }
+        public DelegateCommand MovePrevCommand { get; set; }
+        public DelegateCommand CloseGalleryCommand { get; set; }
+        #endregion
+
+        public GalleryViewModel(DynamoViewModel dynamoViewModel) 
+        {
+            dvm = dynamoViewModel;
+            var pathManager = dynamoViewModel.Model.PathManager;
+            var galleryFilePath = pathManager.GalleryFilePath;
+            var galleryDirectory = pathManager.GalleryDirectory;
+
+            DynamoVersion = string.Format(Properties.Resources.GalleryDynamoVersion,
+                            pathManager.MajorFileVersion,
+                            pathManager.MinorFileVersion);
+
+            contents = GalleryContents.Load(galleryFilePath).GalleryUiContents;
+
+            if (contents != null)
+            {
+                //Set image path relative to gallery Directory
+                foreach (GalleryContent content in contents)
+                {
+                    content.ImagePath = Path.Combine(galleryDirectory, content.ImagePath);
+                }
+
+                currentContent = contents.FirstOrDefault();
+
+                if (currentContent != null) //if contents is not empty
+                {
+                    currentContent.IsCurrent = true;
+                    isAnyContent = true;
+                }
+
+                MoveNextCommand = new DelegateCommand(MoveNext, CanMoveNext);
+                MovePrevCommand = new DelegateCommand(MovePrev, CanMovePrev);
+                CloseGalleryCommand = new DelegateCommand(CloseGallery,CanCloseGallery);
+            }
+        }
+
+        #region event handlers
+        internal event RequestCloseGalleryHandler RequestCloseGallery;
+        internal virtual void OnRequestCloseGallery()
+        {
+            if (RequestCloseGallery != null)
+            {
+                RequestCloseGallery();
+            }
+        }
+
+        internal void CloseGallery(object parameters)
+        {
+            //forward CloseGallery to DynamoViewModel
+            dvm.CloseGalleryCommand.Execute(null);
+        }
+
+        internal bool CanCloseGallery(object parameters)
+        {
+            return true;
+        }
```

Comment:
Same here, we can remove `CanCloseGallery`, `CanMoveNext` and `CanMovePrev`, replacing them with `() => true` in the `DelegateCommand`.